tests/regression/rhbz-1584461/task.yaml

```diff
@@ -0,0 +1,46 @@
+summary: Verify directories from the base snap are not mounted over symlinks in /etc
+
+description: |
+  The host's /etc is made avaialable inside the snap's mount namespace. During
+  setup, snap-confine attempts to preserve some entries from the base snap to
+  present a uniform environment for the snaps. When the entries in host
+  namespace are symlinks, they are followed when mounting and the mount, with
+  source from the base snap, appears in the host's mount namespace.
+
+# we only know of this happening on Fedora, see https://bugzilla.redhat.com/show_bug.cgi?id=1584461
+systems: [fedora-*]
+
+restore: |
+    if [ "$(readlink /etc/ssl)" == "/etc/ssl-backup" ]; then
+        rm /etc/ssl
+        mv /etc/ssl-backup /etc/ssl
+    fi
+
+    rm -f ./in-snap ./on-host
+
+execute: |
+    # on Fedora, /etc/nsswitch.conf is a symlink to /etc/authselect/nsswitch.conf
+    # TODO: uncomment once we fix this in spread images
+    # test -s /etc/nsswitch.conf
```

Comment:
Did you mean 'test -h /etc/nsswitch.conf'?

Reply:
Heh, yes! Thanks for spotting this.